src/sickle.ts

```diff
@@ -261,15 +265,19 @@ class Annotator {
   private writeTextFromOffset(from: number, node: ts.Node): number {
     let to = node.getFullStart();
     if (from == to) return to;
-    assert(to > from, 'Offset must not be smaller');
+    this.assert(to > from, `Offset must not be smaller; ${to} vs ${from}`);
     return this.writeRange(from, to);
   }
 
-  private fail(msg: string) { throw new Error(msg); }
-}
+  private fail(msg: string) {
+    let offset = this.currentNode.getFullStart();
+    let {line, character} = this.file.getLineAndCharacterOfPosition(offset);
+    throw new Error(`near node starting at ${line+1}:${character+1}: ${msg}`);
```

Comment:
I was thinking whether we should add a test for this, but I don't see yet a failure / assertion that would be useful to the external user of the tool. All the errors we have are inconsistencies in our internal state, as which point you can argue it is not worth the overhead of testing that. What do you think?